packages/nvidia_nat_llama_index/src/nat/plugins/llama_index/llm.py

```diff
@@ -13,90 +13,97 @@
 from nat.llm.aws_bedrock_llm import AWSBedrockModelConfig
 from nat.llm.azure_openai_llm import AzureOpenAIModelConfig
 from nat.llm.nim_llm import NIMModelConfig
 from nat.llm.openai_llm import OpenAIModelConfig
+from nat.llm.utils.thinking import BaseThinkingInjector
+from nat.llm.utils.thinking import FunctionArgumentWrapper
+from nat.llm.utils.thinking import patch_with_thinking
 from nat.utils.exception_handlers.automatic_retries import patch_with_retry
+from nat.utils.type_utils import override
 
+ModelType = TypeVar("ModelType")
 
-@register_llm_client(config_type=AWSBedrockModelConfig, wrapper_type=LLMFrameworkEnum.LLAMA_INDEX)
-async def aws_bedrock_llama_index(llm_config: AWSBedrockModelConfig, _builder: Builder):
 
-    from llama_index.llms.bedrock import Bedrock
+def _patch_llm_based_on_config(client: ModelType, llm_config: LLMBaseConfig) -> ModelType:
 
-    kwargs = llm_config.model_dump(exclude={"type", "max_tokens"}, by_alias=True)
+    from llama_index.core.base.llms.types import ChatMessage
 
-    llm = Bedrock(**kwargs)
+    class LlamaIndexThinkingInjector(BaseThinkingInjector):
+
+        @override
+        def inject(self, messages: Sequence[ChatMessage], *args, **kwargs) -> FunctionArgumentWrapper:
+            new_messages = [ChatMessage(role="system", content=self.system_prompt)] + list(messages)
+            return FunctionArgumentWrapper(new_messages, *args, **kwargs)
+
+    if isinstance(llm_config, ThinkingMixin) and llm_config.thinking_system_prompt is not None:
+        client = patch_with_thinking(
+            client,
+            LlamaIndexThinkingInjector(
+                system_prompt=llm_config.thinking_system_prompt,
+                function_names=[
+                    "chat",
+                    "stream_chat",
+                    "achat",
+                    "astream_chat",
+                ],
+            ))
 
     if isinstance(llm_config, RetryMixin):
-        llm = patch_with_retry(llm,
-                               retries=llm_config.num_retries,
-                               retry_codes=llm_config.retry_on_status_codes,
-                               retry_on_messages=llm_config.retry_on_errors)
+        client = patch_with_retry(client,
+                                  retries=llm_config.num_retries,
+                                  retry_codes=llm_config.retry_on_status_codes,
+                                  retry_on_messages=llm_config.retry_on_errors)
+
+    return client
+
+
+@register_llm_client(config_type=AWSBedrockModelConfig, wrapper_type=LLMFrameworkEnum.LLAMA_INDEX)
+async def aws_bedrock_llama_index(llm_config: AWSBedrockModelConfig, _builder: Builder):
+
+    from llama_index.llms.bedrock import Bedrock
 
-    yield llm
+    # LlamaIndex uses context_size instead of max_tokens
+    llm = Bedrock(**llm_config.model_dump(exclude={"type", "top_p"}, by_alias=True), )
+
+    yield _patch_llm_based_on_config(llm, llm_config)
```

Comment:
Would it be better to apply the `if isinstance(llm_config, ThinkingMixin)` check here and potentially avoid the call to `_patch_llm_based_on_config`?